avocado/core/task/statemachine.py

```diff
@@ -6,8 +6,9 @@
 class TaskStateMachine:
     """Represents all phases that a task can go through its life."""
 
-    def __init__(self, tasks):
+    def __init__(self, tasks, status_repo):
```

Comment:
With this change, shouldn't the following locations be changed too?

 * `plugins/runner_nrunner.py:199`
 * `examples/nrunner/task-state-machine.py`

Reply:
I just looked at `examples/nrunner/task-state-machine.py`, and it is just an example of how the state machine works, but it does not have relation to the implementation of the state machine in the Avocado core.

Reply:
Yes, I saw that. But since we are change how our state machine works, IMO would be nice to have a better/updated example.

Reply:
If you run that python script, you will see it already reflects what the state machine does today.